RDMAUpdate/main/Utils/HandlerUtil.py

```diff
@@ -148,13 +149,14 @@ def _parse_config(self, ctxt):
                 thumb = handlerSettings['protectedSettingsCertThumbprint']
                 cert = waagent.LibDir + '/' + thumb + '.crt'
                 pkey = waagent.LibDir + '/' + thumb + '.prv'
-                waagent.SetFileContents('/tmp/kk', protectedSettings)
-                cleartxt = None
-                cleartxt = waagent.RunGetOutput("base64 -d /tmp/kk | openssl smime  -inform DER -decrypt -recip " + cert + "  -inkey " + pkey)[1]
-                os.remove("/tmp/kk")
-                if cleartxt == None:
-                    self.error("OpenSSh decode error using  thumbprint " + thumb)
-                    do_exit(1,operation,'error','1', operation + ' Failed')
+                f = tempfile.NamedTemporaryFile(delete=False)
+                f.close()
+                waagent.SetFileContents(f.name, protectedSettings)
+                cleartxt = waagent.RunGetOutput("base64 -d " + f.name + " | openssl smime  -inform DER -decrypt -recip " + cert + "  -inkey " + pkey)[1]
```

Comment:
I just applied the code (that uses stdin instead of a temp file at all) you provided at that time. It's in the common Utils (top-level) directory, and this appears to be a copy of that directory from an old time...